### PR TITLE
Update entry service with hint field

### PR DESCRIPTION
**Pull request by guilhermocc**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**
Spire server entry service
- BatchUpdateEntry
- BatchCreateEntry
- ListEntries
- GetEntry

**Description of change**
Update spire server entry service:
- Add a new hint filter in the entry listing
- Return hint data in get and list entry
- Accept entry creation with hint data
- Accept entry updates with hint data

**Which issue this PR fixes**
Ongoing work for #2588

### CODE REVIEW

File: pkg/server/api/entry_test.go [hunk 1]
```diff
@@ -184,6 +189,278 @@ func TestProtoToRegistrationEntryWithMask(t *testing.T) {
...
+				Admin:          true,
+				EntryExpiry:    expiresAt,
+				DnsNames:       []string{"dns1", "dns2"},
+				Downstream:     true,
+				RevisionNumber: 99,
+				Hint:           "external",
+			},
+			mask: protoutil.AllTrueEntryMask,
+			err:  "selector list is empty",
+		},
```
Comment (guilhermocc) [inline pkg/server/api/entry_test.go:434]: Extra coverage, not related to the main change

File: pkg/common/telemetry/names.go [hunk 2]
```diff
@@ -271,6 +271,9 @@ const (
 	// ExpiresAt tags registration entry expiration
 	ExpiresAt = "expires_at"
 
+	// Hint tags entry hint
```
Comment (MarcosDY) [inline pkg/common/telemetry/names.go:274]:
```suggestion
	// Hint tags registration entry hint
```

**Review by MarcosDY — COMMENTED** [review]
looks great! A simple nit

File: pkg/common/telemetry/names.go (hunk 2 above)
Comment (guilhermocc) [reply pkg/common/telemetry/names.go]: Thank you for the review, @MarcosDY! [401a07e](https://github.com/spiffe/spire/pull/3926/commits/401a07edcec1ce80d2c600507906ebab6bb26001)

**Review by MarcosDY — APPROVED** [review]
LGTM!

**Pull request merged**